tests/run-pass/crate_traversal.rs

```diff
@@ -1,16 +1,164 @@
-// A small trait to test crate traversal
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+//
 
-trait TestTrait {
-    fn test_method() {}
+// statements and expressions that exercise all visitor code paths
+
+#![allow(const_err)]
+#![allow(unused)]
+#![feature(asm)]
+#![feature(box_syntax)]
+#![feature(generators)]
+
+use std::mem;
+
+pub trait TestTrait {
+    fn test_method() {
+        let _loc = [1];
+        let loc2 = [ 0; 512 ];
+    }
 }
 
-struct TestStruct {
-     test_field: i64
+pub struct TestStruct {
+    pub test_field: i64
 }
 
 impl TestStruct {
-    fn test_function() {}
-    fn test_method(&self) {}
+    pub fn test_function() {}
+    pub fn test_method(&self) {}
+}
+
+pub unsafe fn test1() {
+    #[derive(Copy, Clone)]
+    enum Void {}
+    union A { a: (), v: Void }
+    let a = A { a: () };
+    match a.v {
+    }
+}
+
+fn test2() {
+    let z = 0;
+    let _x = {
+        &z
+    };
+}
+
+fn test3() {
```

Comment:
How do these tests work? Is there an expected output file somewhere?

Reply:
At the moment the role of the test file is just to trigger calls to all of the visitor methods. They don't do anything right now, so there is no output.